### PR TITLE
Ct/get cached institution list

### PR DESCRIPTION
**Pull request by Craiting**
Issue: https://universalconnect.atlassian.net/browse/UCP-58

Sets up the endpoint to get all institutions in the cachedInstitution format.

### CODE REVIEW

File: apps/institution-service/.env.example [hunk 1]
```diff
@@ -0,0 +1,10 @@
+AUTH0_AUDIENCE=
+AUTH0_DOMAIN=
+
+# Cypress testing variables
+E2E_CLIENT_ID=
```
Comment (wesrisenmay-mx) [inline apps/institution-service/.env.example:5]: We should keep cypress out of our main env files. 

Example: https://github.com/Universal-Connect-Project/ucp-hosted-app/blob/main/apps/ucp-ui/cypress.example.env.json

File: apps/institution-service/controllers/institutionController.ts [hunk 2]
```diff
@@ -9,4 +10,35 @@ export const getAllInstitutions = async (req: Request, res: Response) => {
...
+
+    const cachedInstitutionList = institutions.map(
+      transformInstitutionToCachedInstitution
+    );
+
+    res.status(200);
+    res.json(cachedInstitutionList);
+  } catch (error) {
+    console.log("error", error);
+    res.status(400).json({ error: "Error getting all Institutions" });
```
Comment (wesrisenmay-mx) [inline apps/institution-service/controllers/institutionController.ts:42]: Need tests for this

File: apps/institution-service/cypress/e2e/express.cy.ts [hunk 3]
```diff
@@ -1,12 +1,23 @@
-import { PORT } from "./../../shared/const";
-
 describe("Express", () => {
   it("returns pong", () => {
-    cy.request(`http://localhost:${PORT}/ping`).then(
+    cy.request(`http://localhost:8088/ping`).then(
```
Comment (wesrisenmay-mx) [inline apps/institution-service/cypress/e2e/express.cy.ts:3]: Why this change?

File: apps/institution-service/openApi.json [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      "bearerAuth": {
+        "type": "http",
+        "scheme": "bearer",
+        "bearerFormat": "JWT"
+      }
+    }
+  },
+  "servers": [
+    {
+      "description": "Local development widget",
```
Comment (wesrisenmay-mx) [inline apps/institution-service/openApi.json:19]: These 2 lines need updated

File: apps/institution-service/.env.example [hunk 5]
```diff
@@ -0,0 +1,10 @@
+AUTH0_AUDIENCE=
```
Comment (wesrisenmay-mx) [inline apps/institution-service/.env.example:1]: I don't think these 2 variables are sensitive, and they're required for this service to function

Comment (wesrisenmay-mx) [reply apps/institution-service/.env.example:1]: I think we should move them out of the env and hardcode them in a variable somewhere

File: apps/institution-service/.env.example (hunk 1 above)
Comment (Craiting) [reply apps/institution-service/.env.example:5]: I saw that and thought it was kinda confusing because you look at the [cypress.config.ts](https://github.com/Universal-Connect-Project/ucp-hosted-app/blob/89cabb8f1799da3fca0f2bd67326b5edb86c96df/apps/institution-service/cypress.config.ts#L6-L11) and it looks like all the env variables are coming from the .env file. And then locally you have a cypress.example.env which overwrites those variables. But on the github that file doesn't exist so it just pulls the variables from the environment variables.

File: apps/institution-service/cypress/e2e/express.cy.ts (hunk 3 above)
Comment (Craiting) [reply apps/institution-service/cypress/e2e/express.cy.ts:3]: Idk why I changed that, I'll revert it

File: apps/institution-service/.env.example (hunk 5 above)
Comment (Craiting) [reply apps/institution-service/.env.example:1]: Are you trying to say it should be hardcoded everywhere it's used? I could see these variables potentially being different for different environments so it makes sense to me to have them here.

File: apps/institution-service/cypress/support/e2e.ts [hunk 6]
```diff
@@ -14,4 +14,46 @@
...
+      client_id,
+      client_secret,
+    },
+  }).then((response: Cypress.Response<JwtPayload>) => {
+    Cypress.env("ACCESS_TOKEN", response.body.access_token);
+  });
+
+  const username = Cypress.env("E2E_USERNAME") as string;
+  const password = Cypress.env("E2E_PASSWORD") as string;
+  const noAccessClientId = Cypress.env("E2E_CLIENT_ID") as string;
```
Comment (wesrisenmay-mx) [inline apps/institution-service/cypress/support/e2e.ts:40]: Is this the web ui client id and client secret? If so, then we should call it that

File: apps/institution-service/cypress/support/e2e.ts [hunk 7]
```diff
@@ -14,4 +14,46 @@
...
+  const noAccessClientSecret = Cypress.env("E2E_CLIENT_SECRET") as string;
+  const userAudience = Cypress.env("AUTH0_AUDIENCE") as string;
```
Comment (wesrisenmay-mx) [inline apps/institution-service/cypress/support/e2e.ts:42]: Does this fail if you use the auth0_widget_audience?

File: apps/institution-service/cypress/support/e2e.ts (hunk 6 above)
Comment (Craiting) [reply apps/institution-service/cypress/support/e2e.ts:40]: I'm not sure but I can figure it out. Do you want the env variable to be renamed or just the local variable?

Comment (Craiting) [reply apps/institution-service/cypress/support/e2e.ts:40]: Just checked and yes it is the UCP Web UI Client

File: apps/institution-service/cypress/support/e2e.ts (hunk 7 above)
Comment (Craiting) [reply apps/institution-service/cypress/support/e2e.ts:42]: Yeah

File: apps/institution-service/cypress/support/e2e.ts (hunk 6 above)
Comment (wesrisenmay-mx) [reply apps/institution-service/cypress/support/e2e.ts:40]: Yeah I would rename them all to web ui

**Review by wesrisenmay-mx — APPROVED** [review]
Nice work
